src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -133,6 +133,33 @@ preflight1(PreflightContext const& ctx)
 NotTEC
 preflight2(PreflightContext const& ctx)
 {
+    if (ctx.flags & tapDRY_RUN)  // simulation
+    {
+        if (!ctx.tx.getSignature().empty())
+        {
+            // NOTE: This code should never be hit because it's checked in the
+            // `simulate` RPC
+            return temINVALID;  // LCOV_EXCL_LINE
+        }
+
+        if (!ctx.tx.isFieldPresent(sfSigners))
+        {
+            // no signers, no signature - a valid simulation
+            return tesSUCCESS;
+        }
+
+        for (auto const& signer : ctx.tx.getFieldArray(sfSigners))
+        {
+            if (signer.isFieldPresent(sfTxnSignature) &&
+                !ctx.tx.getSignature().empty())
```

Comment:
`!signer[sfTxnSignature].empty()`, ye?

Reply:
Fixed in [f1ac9b8](https://github.com/XRPLF/rippled/pull/5069/commits/f1ac9b8dc444105d8227afbb6089c9486c351500)